src/util/locking/RedisLocker.ts

```diff
@@ -53,8 +62,12 @@ export class RedisLocker implements ReadWriteLocker, ResourceLocker, Initializab
   private readonly attemptSettings: Required<AttemptSettings>;
   private finalized = false;
 
-  public constructor(redisClient = '127.0.0.1:6379', attemptSettings: AttemptSettings = {}) {
-    this.redis = this.createRedisClient(redisClient);
+  public constructor(
+    redisClient = '127.0.0.1:6379',
+    attemptSettings: AttemptSettings = {},
+    redisOptions: RedisOptions = {},
+  ) {
+    this.redis = this.createRedisClient(redisClient, redisOptions);
```

Comment:
Now you could do something like this:
```ts
const { namespacePrefix, ...options } = redisSettings;
this.namespacePrefix = namespacePrefix; // This could it on the main branch, hence the request for a rebase
this.redis = this.createRedisClient(redisClient, options);
```